src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -1520,10 +1545,26 @@ def to_series(self) -> None:
         Index.to_frame : Convert an Index to a DataFrame.
         Series.to_frame : Convert Series to DataFrame.
         """
-        # TODO: SNOW-1458117 implement to_series
+        from snowflake.snowpark.modin.pandas import Series
 
-    @index_not_implemented()
-    def to_frame(self) -> None:
+        if name is None:
+            name = self.name
+        new_qc = (
+            self._query_compiler.drop(columns=self._query_compiler.columns)
+            .reset_index()
+            .set_index(["index"], drop=False)
+            .set_index_names([None])
+        )
+        if index is not None:
+            new_qc = new_qc.set_index_from_series(Series(index)._query_compiler)
```

Comment:
we should have a way to set_index from index. Can you double check? If not, please put a todo with https://snowflakecomputing.atlassian.net/browse/SNOW-1458137, we should make sure implement it there.

Reply:
a series does not have a set_index method. For us to change a series index, we can do `ser.index = my_index`